pwiz_tools/Skyline/Model/AuditLog/AuditLogEntry.cs

```diff
@@ -265,21 +320,20 @@ public static bool ReadFromXmlTextReader(XmlTextReader reader, out string logged
                 .ChangeActualHash(result.CalculateRootHash())
                 .ChangeSkylHash(Hash.FromBase64(rootHash));
 
+            // yell if audit log entry hash does not match.
             // If the docFormat is null, this is an old audit log and there won't be any entry hashes.
             // We can't just always ignore non-existent hashes, otherwise people could just delete the hash elements
             // and get Skyline to successfully load the audit log
-            if (docFormat != null && !result.RootHash.SkylAndActualHashesEqual())
+            var modifiedEntries =
+                result.AuditLogEntries.Enumerate().Where(entry => !entry.Hash.SkylAndActualHashesEqual());
+
+            if (docFormat != null && (!result.RootHash.SkylAndActualHashesEqual() || modifiedEntries.Count() > 0))
             {
-                var modifiedEntries =
-                    result.AuditLogEntries.Enumerate().Where(entry => !entry.Hash.SkylAndActualHashesEqual());
-                var dlg = new AlertDlg(AuditLogStrings.AuditLogList_ReadFromFile_The_following_audit_log_entries_were_modified +
-                                       string.Join(Environment.NewLine,
-                                           modifiedEntries.Select(entry => entry.UndoRedo.ToString())),
-                    MessageBoxButtons.OK);
-                dlg.ShowParentlessDialog();
+                throw new AuditLogException(
+                    AuditLogStrings.AuditLogList_ReadFromFile_The_following_audit_log_entries_were_modified +
+                    string.Join(Environment.NewLine, modifiedEntries.Select(entry => entry.UndoRedo.ToString())));
```

Comment:
There's a method called "TextUtil.LineSeparate" which you might want to use instead of string.Join